hw/ip/spi_device/data/spi_device.hjson

```diff
@@ -669,6 +677,259 @@
+        } // f: rdfifo_depth
+        { bits: "10:8"
+          name: "wrfifo_depth"
+          desc: "This field represents the current write FIFO depth."
+        } // f: wrfifo_depth
+      ]
+    } // R: TPM_STATUS
+    { multireg: {
+      cname: "TPM"
+      name: "TPM_ACCESS"
+      desc: '''TPM_ACCESS_x register.
+        '''
+      count: "NumLocality"
+      compact: true
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "7:0"
+          name: "access"
+          desc: "TPM_ACCESS"
+        }
+      ]
+    }} // mr: TPM_ACCESS
+    { name: "TPM_STS"
+      desc: '''TPM_STS_x register.
+
+        The register is mirrored to all Localities.
+        The value is returned to the host system only when the activeLocality
+        in the TPM_ACCESS_x is matched to the current received Locality.
+        '''
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "31:0"
+          name: "sts"
+          desc: "TPM_STS_x"
+        }
+      ]
+    } // R: TPM_STS
+    { name: "TPM_INTF_CAPABILITY"
+      desc: "TPM_INTF_CAPABILITY"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "31:0"
+          name: "intf_capability"
+          desc: "TPM_INTF_CAPABILITY"
+        }
+      ]
+    } // R: TPM_INTF_CAPABILITY
+    { name: "TPM_INT_ENABLE"
+      desc: "TPM_INT_ENABLE"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "31:0"
+          name: "int_enable"
+          desc: "TPM_INT_ENABLE"
+        }
+      ]
+    } // R: TPM_INT_ENABLE
+    { name: "TPM_INT_VECTOR"
+      desc: "TPM_INT_VECTOR"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "7:0"
+          name: "int_vector"
+          desc: "TPM_INT_VECTOR"
+        }
+      ]
+    } // R: TPM_INT_VECTOR
+    { name: "TPM_INT_STATUS"
+      desc: "TPM_INT_STATUS"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "31:0"
+          name: "int_status"
+          desc: "TPM_INT_STATUS"
+        }
+      ]
+    } // R: TPM_INT_STATUS
+    { name: "TPM_DID_VID"
+      desc: "TPM_DID/ TPM_VID register"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "15:0"
+          name: "vid"
+          desc: "TPM_VID"
+        } // f: vid
+        { bits: "31:16"
+          name: "did"
+          desc: "TPM_DID"
+        } // f: did
+      ]
+    } // R: TPM_DID_VID
+    { name: "TPM_RID"
+      desc: "TPM_RID"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "7:0"
+          name: "rid"
+          desc: "TPM_RID"
+        }
+      ]
+    } // R: TPM_RID
+    { name: "TPM_CMD_ADDR"
+      desc: '''TPM Command and Address buffer
+
+        The SW may get the received TPM command and address by readin gthis CSR.
+        '''
+      swaccess: "ro"
+      hwaccess: "hrw"
+      hwext: "true"
+      hwre:  "true"
+      hwqe:  "true"
+      fields: [
+        { bits: "23:0"
```

Comment:
so is the idea is that software would use the address to determine which locality is accessed?

Reply:
Yes.